contracts/interfaces/ISAVAX.sol

```diff
@@ -1,7 +1,11 @@
 // SPDX-License-Identifier: MIT
 pragma solidity ^0.8.4;
 
-interface ISAVAX {
+import {
+    IERC20Upgradeable
+} from "@openzeppelin/contracts-upgradeable/token/ERC20/IERC20Upgradeable.sol";
+
+interface ISAVAX is IERC20Upgradeable {
```

Comment:
Does it it need to be IERC20Upgradeable? My understanding is that it is a plain IERC20

Reply:
It's a proxy:
https://snowtrace.io/address/0x2b2c81e08f1af8835a78bb2a90ae924ace0ea4be#code

Here it's `IERC20Upgradeable` (line 1346):

```
contract StakedAvax is
    IERC20Upgradeable,
    AccessControlUpgradeable,
    ReentrancyGuardUpgradeable,
    PausableUpgradeable,
    StakedAvaxStorage
{
```

Reply:
there's no need to extend `IERC20Upgradeable` , in fact we only need this function for the sAVAX-specific transactions

```
    function submit() external payable returns (uint256);
```

Reply:
Seems like we need an `approve` method here: https://github.com/ribbon-finance/ribbon-v2/pull/226/files#r816417713

Reply:
@shibacrypt `approve()` is inferred from the `IERC20Upgradeable` interface